realm-jni/src/io_realm_internal_CheckedRow.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io_realm_internal_CheckedRow.h"
+#include "io_realm_internal_UncheckedRow.h"
+
+#include "util.hpp"
+#include "mixedutil.hpp"
+#include "tablebase_tpl.hpp"
+
+using namespace realm;
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnCount
+  (JNIEnv* env, jobject obj, jlong nativeRowPtr)
+{
+    if (!ROW(nativeRowPtr)->is_attached())
+        return 0;
+
+    return Java_io_realm_internal_UncheckedRow_nativeGetColumnCount(env, obj, nativeRowPtr);
+}
+
+JNIEXPORT jstring JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnName
+  (JNIEnv* env, jobject obj, jlong nativeRowPtr, jlong columnIndex)
+{
+    if (!ROW_AND_COL_INDEX_VALID(env, ROW(nativeRowPtr), columnIndex))
+        return NULL;
+
+    return Java_io_realm_internal_UncheckedRow_nativeGetColumnName(env, obj, nativeRowPtr, columnIndex);
+}
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_CheckedRow_nativeGetColumnIndex
+  (JNIEnv* env, jobject obj, jlong nativeRowPtr, jstring columnName)
+{
+    if (!ROW(nativeRowPtr)->is_attached())
+        return 0;
+
+    jlong ndx = Java_io_realm_internal_UncheckedRow_nativeGetColumnIndex(env, obj, nativeRowPtr, columnName);
+    if (ndx == realm::not_found) {
```

Comment:
I'm not sure if it's a problem but `realm::not_found` is `size_t(-1)` while `ndx` will have the value `jlong(-1)`. `jlong` is always a 64 bit signed integer while `size_t` is unsigned and depends on bit width.